src/utils/set.ts

```diff
@@ -24,17 +28,38 @@ module Plottable {
         var index = this._values.indexOf(value);
         if (index !== -1) {
           this._values.splice(index, 1);
+          this._setSize(this._values.length);
           return true;
         }
         return false;
       }
 
+      private _setSize(size: number) {
+        Object.defineProperty(this, "size", {
+          value: this._values.length,
+          configurable: true,
+          writable: false
```

Comment:
According to this
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object/defineProperty

`Object.defineProperty` defaults to set immutable values and thus the `writable` parameter is default to false and this explicit setting can be removed